### PR TITLE
v2

### PR DESCRIPTION
**Pull request by balloob**
- Don't convert to kW / kWh but leave data as we get it from API **Breaking**
- Fix bug calculating difference by resetting `last_value` to 0 when calculating diffs
- Create re-usable methods to back properties
- Add tests

### CODE REVIEW

**Review by frenck — CHANGES_REQUESTED** [review]
We should rename the `kwh*` variables to `wh*`, just to prevent confusion for the future

**Comment by balloob** [comment]
Done.

**Review by frenck — APPROVED** [review]
Nice! 👍

File: tests/test_models.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    assert estimate.sum_energy_production(12) == 13454
+    assert estimate.sum_energy_production(24) == 13454
+
+
+def test_estimate_near_end(patch_near_end_today):
+    """Test estimate."""
+    estimate = models.Estimate.from_dict(PAYLOAD)
+
+    assert estimate.timezone == "Europe/Amsterdam"
+    print("NOW", estimate.now())
```
Comment (MartinHjelmare) [inline tests/test_models.py:63]: Leftover debug print?

Comment (balloob) [reply tests/test_models.py:63]: a320a9f

**Pull request merged**